src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -276,9 +284,25 @@ private static string GenerateUniqueChannelLocalName()
 
                 lock (_sessionGuard)
```

Comment:
This is incorrect. The lock will not be held while the task continuation (almost the entire method) is executed.